.github/workflows/ci_bench_block_execution.yaml

```diff
@@ -0,0 +1,101 @@
+name: Benchmark Block execution against main
+
+on:
+  pull_request:
+    branches: ["**"]
+
+jobs:
+  build-binaries:
+    strategy:
+      matrix:
+        branch: [ base, head ]
+    name: Build ethrex for ${{ matrix.branch }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Populate cache
+        uses: actions/cache@v4
+        id: cache
+        with:
+          path: bin/ethrex-${{ matrix.branch }}
+          key: binary-${{ github.event.pull_request[matrix.branch].sha }}
+  
+      - name: Install Rust
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        uses: dtolnay/rust-toolchain@stable
+  
+      - name: Checkout
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request[matrix.branch].sha }}
+  
+      - name: Fetch Rust cache
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        uses: Swatinem/rust-cache@v2
+  
+      - name: Build binary
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        run: |
+          cargo b --release --bin ethrex
+          mkdir bin
+          cp target/release/ethrex bin/ethrex-${{ matrix.branch }}
+
+  run-hyperfine:
+    name: Run benchmark against main
+    needs: [ build-binaries ]
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout PR
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: Install Hyperfine
+      uses: taiki-e/install-action@v2
+      with:
+        tool: hyperfine@1.16
+
+    - name: Fetch base binary
+      uses: actions/cache/restore@v3
+      with:
+        path: bin/ethrex-base
+        key: binary-${{ github.event.pull_request.base.sha }}
+
+    - name: Fetch HEAD binary
+      uses: actions/cache/restore@v3
+      with:
+        path: bin/ethrex-head
+        key: binary-${{ github.event.pull_request.head.sha }}
+
+    - name: Benchmark against main
+      id: run-benchmarks
+      run: |
+        sudo swapoff -a
+        BINS="head"
+        hyperfine -w 5 -N -r 10 --show-output --export-markdown "bench_pr_comparison.md" \
+        -L bin "$BINS" -n "{bin}" \
+        "./bin/ethrex-{bin} --network test_data/genesis-l2.json --import ./test_data/l2-loadtest.rlp --only_import"
```

Comment:
`l2` nomenclature is a bit weird imo, this is pure l1 stuff.

Reply:
Agreed, the `l2-loadtest.rlp` and `genesis-l2.json` files here will be replaced by different ones before merging the PR